stem/output.py

```diff
@@ -0,0 +1,219 @@
+import warnings
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Dict, Any
+from enum import Enum
+
+DEFAULT_GID_FLAGS = {
+    "WriteDeformedMeshFlag": "WriteUndeformed",
+    "WriteConditionsFlag": "WriteElementsOnly",
+    "GiDPostMode": "GiD_PostBinary",
+    "MultiFileFlag": "SingleFile",
+}
+
+
+class NodalOutput(Enum):
+    DISPLACEMENT = 1
+    DISPLACEMENT_X = 11
+    DISPLACEMENT_Y = 12
+    DISPLACEMENT_Z = 13
+    VELOCITY = 2
+    VELOCITY_X = 21
+    VELOCITY_Y = 22
+    VELOCITY_Z = 23
+    ACCELERATION = 3
+    ACCELERATION_X = 31
+    ACCELERATION_Y = 32
+    ACCELERATION_Z = 33
+    TOTAL_DISPLACEMENT = 4
+    TOTAL_DISPLACEMENT_X = 41
+    TOTAL_DISPLACEMENT_Y = 42
+    TOTAL_DISPLACEMENT_Z = 43
+    WATER_PRESSURE = 5
+    VOLUME_ACCELERATION = 6
+    VOLUME_ACCELERATION_X = 61
+    VOLUME_ACCELERATION_Y = 62
+    VOLUME_ACCELERATION_Z = 63
+
+
+TENSOR_OUTPUTS = [
+    "GREEN_LAGRANGE_STRAIN",
+    "ENGINEERING_STRAIN",
+    "CAUCHY_STRESS",
+    "TOTAL_STRESS",
+]
+
+
+def detect_tensor_outputs(requested_outputs: List[str]):
+    """
+    Detects whether gauss point outputs are requested and warns the user
+
+    Args:
+        requested_outputs (List[st]): list of requested outputs (gauss point)
+    """
+    if len(requested_outputs) > 0:
+        detected_tensor_outputs = []
+        for requested_output in requested_outputs:
+            for tensor_output in TENSOR_OUTPUTS:
+                if tensor_output in requested_output:
+                    detected_tensor_outputs.append(tensor_output)
+                    break
+        if len(detected_tensor_outputs):
+            _fmt_list = "".join([f" - {outpt} \n" for outpt in detected_tensor_outputs])
+            _msg = (
+                f"\n[WARNING] tensor output detected:\n{_fmt_list}"
+                f"The outputs are incorrectly rendered for the selected output\n"
+                f"if vector is selected and ignored if tensor is selected."
+            )
+            print(_msg)
+
+
+@dataclass
+class OutputParametersABC(ABC):
+    """
+    Abstract class for the definition of user output parameters (GiD, VTK, json).
+
+    Attributes:
+        -
+    """
+
+    pass
+
+    @abstractmethod
+    def validate(self):
+        """
+        Abstract method for validating user inputs
+        """
+        pass
+
+
+@dataclass
+class GiDOutputParameters(OutputParametersABC):
```

Comment:
Can you extende all docstrings with a description of the attributes. And also indicate the possible options in case it is required. For example: output_control_type should be time or step

Reply:
This will take some times as I don't actually know all the options for KRATOS